### PR TITLE
Search vowels in begginer mode

### PR DESCRIPTION
**Pull request by yvalenta**
- [X] Search vowels in begginer mode
  I want to improve slowly

:dash:

### CODE REVIEW

File: yvalenta/search.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  end
+
+  def open_file
+    f = File.open("../text.txt", "r")
+    run(f)
+    f.close
+  end
+
+  def time_start(start)
+    struct = Struct.new(:start)
```
Comment (rderoldan1) [inline yvalenta/search.rb]: Why use a struct?

File: yvalenta/search.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    @time_s = struct.new(start)
+  end
+
+  def time_finish(finish)
+    struct = Struct.new(:finish)
```
Comment (rderoldan1) [inline yvalenta/search.rb]: Why use a struct?

File: yvalenta/search.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        validate(c)
+      end
+    end
+    print(vowels)
+  end
+
+  def validate char
+    if !@vowels[char].nil?
+      @vowels[char] += 1
+      @count += 1
```
Comment (rderoldan1) [inline yvalenta/search.rb]: Para que este @count?

File: yvalenta/search.rb [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  private
+
+  def vowels
+    @vowels ||= { 'a' => 0, 'e' => 0, 'i' => 0, 'o' => 0, 'u' => 0 }
+  end
+
+  def run file
+    time_start Time.now
+    file.each_line do |line|
+      line.downcase.split("").each do |c|
```
Comment (rderoldan1) [inline yvalenta/search.rb]: el problema de este split es el siguiente:

``` ruby
"oe".split.count => 2
"una linea mas larga".split.count => 16
"una linea que tiene 10000 letras".split.count => 100000
```

En vez de esto debe usar una función que solo permita iterar sobre las vocales

File: yvalenta/search.rb (hunk 1 above)
Comment (yvalenta) [inline yvalenta/search.rb]: For burning face,
but i would like to know in which cases can be applied struct

File: yvalenta/search.rb (hunk 2 above)
Comment (yvalenta) [inline yvalenta/search.rb]: vis..

File: yvalenta/search.rb (hunk 1 above)
Comment (rderoldan1) [inline yvalenta/search.rb]: Is nice to see you learning, so maybe, you learnt that in this case is not useful!!

File: yvalenta/search.rb (hunk 2 above)
Comment (rderoldan1) [inline yvalenta/search.rb]: Is nice to see you learning, so maybe, you learnt that in this case is not useful!!

File: yvalenta/search.rb (hunk 3 above)
Comment (yvalenta) [inline yvalenta/search.rb]: to count the number of characters in the text, but i delete the line that prints

File: yvalenta/search.rb (hunk 4 above)
Comment (yvalenta) [inline yvalenta/search.rb]: Esa es la actitud explicar con plastilina ..

File: yvalenta/search.rb (hunk 3 above)
Comment (rderoldan1) [inline yvalenta/search.rb]: If it don't have any function, please, remove it

File: yvalenta/search.rb (hunk 4 above)
Comment (rderoldan1) [inline yvalenta/search.rb]: Adicional, tiene otro problema: 

``` ruby
"oe".split.each => 2 iteraciones, 2 vocales
"una linea mas larga".split.each => 16 iteraciones, 7 vocales
"una linea que tiene 10000 letras".split.each => 10000 iteraciones, n vocales
```

Como se ve, el número de vocales siempre es menor al número total de letras, por lo que se demora mas al procesar todas las letras.

File: yvalenta/search.rb [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  end
+
+  def validate char
+    if !@vowels[char].nil?
+      @vowels[char] += 1
+    end
+    time_finish Time.now
+  end
+
+  def print vowels
```
Comment (andru1989) [inline yvalenta/search.rb]: Is not necesary to do a loop for print the vowels, See that you have a method and a global variable, even is useless to receive an argument that you have as a global variable

File: yvalenta/search.rb [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  end
+
+  def time_finish(finish)
+    struct = Struct.new(:finish)
+    @time_f = struct.new(finish)
+  end
+
+  private
+
+  def vowels
```
Comment (andru1989) [inline yvalenta/search.rb:16]: I think is better if you define the hash in initialize method, is not necesary to define another method to do it

File: yvalenta/search.rb [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  def print vowels
+    vowels.each do |k,v|
+      p "#{k} => #{v}"
+    end
+    puts "Run in #{(@time_f.finish - @time_s.start)} seconds (around 16 years to execute)"
+  end
+
+end
+
+print SearchVowels.new().open_file
```
Comment (andru1989) [inline yvalenta/search.rb]: Why do you print the output again ?, see that you print it in method too

File: yvalenta/search.rb (hunk 6 above)
Comment (yvalenta) [inline yvalenta/search.rb:16]: according told me @pwalvarado , should not be initialized in the way you say it

what should I do?

**Pull request merged**